metricbeat/beater/metricbeat.go

```diff
@@ -255,6 +255,19 @@ func (bt *Metricbeat) Run(b *beat.Beat) error {
 		}()
 	}
 
+	// Start the manager after all the reload hooks are configured,
+	// the Manager is stopped at the end of the execution.
+	if err := b.Manager.Start(); err != nil {
+		return err
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		<-bt.done
```

Comment:
The main issue, is all the beats have a different bootstrap code, which made it nearly impossible to ensure that the `reload` steps are configured before everything else. After looking more closely in the code, I can say that the boot process of beats is somewhat undefined. I would  not be surprised that other issues exist because of these quirks.


I am not a big fan of having adhoc code in each beat. But this was in my opinion to ensure that the system was in a consistent state before even starting the manager.